metricbeat/module/nifi/flowfilerepostorage/data.go

```diff
@@ -0,0 +1,55 @@
+package flowfilerepostorage
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// FlowFileRepositoryStorageUsage ...
+type FlowFileRepositoryStorageUsage struct {
+	FreeSpace       string `json:"freeSpace"`
+	FreeSpaceBytes  uint64 `json:"freeSpaceBytes"`
+	TotalSpace      string `json:"totalSpace"`
+	TotalSpaceBytes uint64 `json:"totalSpaceBytes"`
+	UsedSpace       string `json:"usedSpace"`
+	UsedSpaceBytes  uint64 `json:"usedSpaceBytes"`
+	Utilization     string `json:"utilization"`
+}
+
+// Data ...
+type Data struct {
+	SystemDiagnostics struct {
+		AggregateSnapshot struct {
+			FlowFileRepositoryStorageUsage `json:"flowFileRepositoryStorageUsage"`
+		} `json:"aggregateSnapshot"`
+	} `json:"systemDiagnostics"`
+}
+
+func eventMapping(body io.Reader) common.MapStr {
+	var data Data
+	err := json.NewDecoder(body).Decode(&data)
+	if err != nil {
+		logp.Err("Error: ", err)
+	}
+
+	slice := data.SystemDiagnostics.AggregateSnapshot.FlowFileRepositoryStorageUsage
+
+	fmt.Printf("%v", data)
+	fmt.Printf("%v", slice)
+
+	event := common.MapStr{
```

Comment:
Yes, that is an improvement we should definitively do in the near future. An other improvement that could help you with this PR and is going in soonish hopefully is here: https://github.com/elastic/beats/pull/3413